src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1316,10 +1326,16 @@ static void OnTitleViewChanged(BindableObject bindable, object oldValue, object
 			var newView = (View)newValue;
 
 			if (oldView != null)
+			{
+				VisualDiagnostics.OnChildRemoved((Page)bindable, oldView, ((IVisualTreeElement)bindable).GetVisualChildren().IndexOf(oldView));
 				oldView.Parent = null;
+			}
 
 			if (newView != null)
+			{
 				newView.Parent = owner;
+				VisualDiagnostics.OnChildAdded((Page)bindable, newView);
```

Comment:
Can we just call AddLogicalChild here?

Reply:
Add/RemoveLogicalChild are on Shell (https://github.com/dotnet/maui/blob/9aa30fc98ebf8b6bd24d8f72c5724baae7f98145/src/Controls/src/Core/Shell/Shell.cs#L707) but `Bindable` could be a ContentPage (At least going by the unit tests that failed when I did cast to Shell (https://github.com/dotnet/maui/pull/14006/commits/fc2775f5caea72ef87d3ad9b0572b416246205e9)).

Seeing that this is in Shell, I could check if Bindable is Shell, and if so, call Add/RemoveLogicalChild, but if it's anything else it would be missed. Is that okay?

Reply:
I think so :-) 

There's two places we need to account for the TitleView being added

1) On the ContentPage
2) On the Shell

If it's added on the ContentPage then we want the `titleView` to get added to the LogicalChildren of the PAge not shell. If it's added to Shell we want it to get added to the LogicalChildren of Shell not Page